src/bun.js/bindings/exports.zig

```diff
@@ -2154,31 +2160,37 @@ pub const ZigConsoleClient = struct {
                 },
                 .Double => {
-                    if (value.isCell()) {
-                        var number_name = ZigString.Empty;
-                        value.getClassName(this.globalThis, &number_name);
+                    if (value.isSymbol()) {
+                        this.addForNewLine("NaN".len);
+                        writer.print(comptime Output.prettyFmt("<r><yellow>NaN<r>", enable_ansi_colors), .{});
+                        return;
+                    }
 
-                        var number_value = ZigString.Empty;
-                        value.toZigString(&number_value, this.globalThis);
+                    // if (value.isCell()) {
+                    //     var number_name = ZigString.Empty;
+                    //     value.getClassName(this.globalThis, &number_name);
 
-                        if (!strings.eqlComptime(number_name.slice(), "Number")) {
-                            this.addForNewLine(number_name.len + number_value.len + "[Number ():]".len);
-                            writer.print(comptime Output.prettyFmt("<r><yellow>[Number ({s}): {s}]<r>", enable_ansi_colors), .{
-                                number_name,
-                                number_value,
-                            });
-                            return;
-                        }
+                    //     var number_value = ZigString.Empty;
+                    //     value.toZigString(&number_value, this.globalThis);
 
-                        this.addForNewLine(number_name.len + number_value.len + 4);
-                        writer.print(comptime Output.prettyFmt("<r><yellow>[{s}: {s}]<r>", enable_ansi_colors), .{
-                            number_name,
-                            number_value,
-                        });
-                        return;
-                    }
+                    //     if (!strings.eqlComptime(number_name.slice(), "Number")) {
+                    //         this.addForNewLine(number_name.len + number_value.len + "[Number ():]".len);
+                    //         writer.print(comptime Output.prettyFmt("<r><yellow>[Number ({s}): {s}]<r>", enable_ansi_colors), .{
+                    //             number_name,
+                    //             number_value,
+                    //         });
+                    //         return;
+                    //     }
+
+                    //     this.addForNewLine(number_name.len + number_value.len + 4);
+                    //     writer.print(comptime Output.prettyFmt("<r><yellow>[{s}: {s}]<r>", enable_ansi_colors), .{
+                    //         number_name,
+                    //         number_value,
+                    //     });
+                    //     return;
+                    // }
 
-                    const num = value.asNumber();
+                    const num = value.coerce(f64, this.globalThis);
```

Comment:
After I dive in the code, it seems coerce won't throw error, it will return NaN if error occurs.

Because of this, when we console.log({ [Symbol.toPrimative]: () => { throw 1 } }), it will print NaN, won't print the error stack as same as node.

See the code here:
https://github.com/oven-sh/bun/blob/5029fc8564f2f2dceef4d83ef4bd7f9ecc17ac72/src/bun.js/bindings/bindings.cpp#L3516-L3527

I have no idea why bun did this.

Reply:
ah i see. `NaN` is better than a crash

Reply:
So, should we keep it? If so, I will add a test for `throw 1` and the result to `NaN`.

Reply:
yeah that sounsd good